### PR TITLE
[exporter/elasticsearch] set the default User-Agent header in the outgoing HTTP requests 

### LINKED ISSUE #29898: [exporter/elasticsearch] Set user-agent header
### Component(s)
exporter/elasticsearch

### Is your feature request related to a problem? Please describe.
The Elasticsearch exporter doesn't set the User-Agent header in the outgoing HTTP requests it makes, but it should.

### Describe the solution you'd like
Following the pattern set in the otlphttpexporter, it should use fmt.Sprintf("%s/%s (%s/%s)", set.BuildInfo.Description, set.BuildInfo.Version, runtime.GOOS, runtime.GOARCH) as the user agent value.

### Describe alternatives you've considered
No response

### Additional context
No response

### PR DESCRIPTION
**Pull request by JaredTan95**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** close https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29898

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by fatsheep9146 — APPROVED** [review]

**Comment by crobert-1** [comment]
Is there any test that covers this functionality, or could ensure that the `User-Agent` is set properly?

**Comment by JaredTan95** [comment]
> Is there any test that covers this functionality, or could ensure that the `User-Agent` is set properly?

nice suggestion, I will update it.

File: exporter/elasticsearchexporter/factory.go [hunk 1]
```diff
@@ -116,10 +116,13 @@ func createTracesExporter(ctx context.Context,
 		exporterhelper.WithQueue(cf.QueueSettings))
 }
 
-// set default User-Agent header
-func setDefaultUserAgentHeader(cf *Config, info component.BuildInfo) {
-	if cf.Headers == nil {
-		cf.Headers = make(map[string]string)
+// set default User-Agent header if User-Agent is empty
+func setDefaultUserAgentHeader(cf *Config, info component.BuildInfo) *Config {
```
Comment (crobert-1) [inline exporter/elasticsearchexporter/factory.go:120]: I'm a bit confused by the combination of a method name `set...` while including a return here: I'd expect a method to either return an updated value, or set the value itself, not both. Is there something I'm missing or can we simplify the method here? 

In other words, the returned value is simply the passed in `cf` variable, so I don't see the point of returning anything.

Comment (JaredTan95) [reply exporter/elasticsearchexporter/factory.go:120]: make sense, there are some confusing semantics

**Review by crobert-1 — APPROVED** [review]

File: exporter/elasticsearchexporter/factory.go [hunk 2]
```diff
@@ -108,3 +115,13 @@ func createTracesExporter(ctx context.Context,
...
 }
+
+// set default User-Agent header with BuildInfo if User-Agent is empty
+func setDefaultUserAgentHeader(cf *Config, info component.BuildInfo) {
+	if _, found := cf.Headers[userAgentHeaderKey]; !found {
+		if cf.Headers == nil {
+			cf.Headers = make(map[string]string)
+			cf.Headers[userAgentHeaderKey] = fmt.Sprintf("%s/%s (%s/%s)", info.Description, info.Version, runtime.GOOS, runtime.GOARCH)
+		}
+	}
```
Comment (bogdandrutu) [inline exporter/elasticsearchexporter/factory.go:126]: I think there is a bug here, if you have Headers with one header (not user-agent) this code will not work). Consider to write the code in a more readable way, to see this:

```golang
	if _, found := cf.Headers[userAgentHeaderKey]; found {
		return
	}
	if cf.Headers == nil {
		cf.Headers = make(map[string]string)
	}
	cf.Headers[userAgentHeaderKey] = fmt.Sprintf("%s/%s (%s/%s)", info.Description, info.Version, runtime.GOOS, runtime.GOARCH)
	}
```

Comment (JaredTan95) [reply exporter/elasticsearchexporter/factory.go:126]: nice catch , I will polish here and add more test to cover it.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**